src/components/NinDisplay.js

```diff
@@ -71,17 +101,9 @@ export class NinDisplay extends Component {
           userData = [
             <div key="1" className="display-nin-show-hide">
               <p key="1" className="display-data verified">
-                { this.props.showNinAtProfile  ? 
-                  this.props.nins[0].number :  
-                  this.props.nins[0].number.replace(/\d{4}$/, '****')
-                }
+                {isShowLastFourDigits(this.props.showNinAtProfile, this.props.nins[0].number)}
               </p>
-             <button key="2" className="show-hide-button" onClick={this.props.toggleShowNinAtProfile}>
-                { this.props.showNinAtProfile ? 
-                  this.props.translate("nin_hide_last_four_digits") : 
-                  this.props.translate("nin_show_last_four_digits")
-                }
-              </button> 
+              {showHideButton("profile", this.props)}
             </div>
```

Comment:
Perhaps it could improve readability to separate the redux hide/show logic from the render logic of any surrounding structures (i.e. streamlining the redux logic to not depend on position and only using position prop in return statement)? I might be wrong, but it looks like all this component needs to know about the nin is if any are saved so you can render for both cases (none: the text/link | any: number and button + separate logic handling what number looks like).